Source/WebCore/crypto/mac/cryptokit/CryptoKitAesShim.h

```diff
@@ -0,0 +1,47 @@
+/*
+* Copyright (C) 2024 Apple Inc. All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions
+* are met:
+* 1. Redistributions of source code must retain the above copyright
+*    notice, this list of conditions and the following disclaimer.
+* 2. Redistributions in binary form must reproduce the above copyright
+*    notice, this list of conditions and the following disclaimer in the
+*    documentation and/or other materials provided with the distribution.
+*
+* THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+* AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+* THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+* PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+* BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+* CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+* THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#pragma once
+
+#import <stdint.h>
+
+namespace WebCore {
+
+class CkAesGcm {
+public:
+    static int encrypt(
```

Comment:
Can we return an error enum instead of just an int?
Do we want to handle any errors?
Can we pass std::span<const uint8_t> here instead of pointer/size?
It seems concerning that we pass a ciphertext pointer but not a size of that.

Reply:
> It seems concerning that we pass a ciphertext pointer but not a size of that.
for AES_GCM plaintext length always is equal to ciphertext length. Just reducing one parameter.

> Do we want to handle any errors?
It's either a pass or a fail for this one. There is nothing to be handled and taken action on. The implementation before does the same.

> Can we pass std::span here instead of pointer/size?
I have removed the shim completely and also now passing in a reference to the original vectors.